pkg/network/usm/utils/connection_rollup.go

```diff
@@ -0,0 +1,284 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package utils
+
+import (
+	"github.com/DataDog/datadog-agent/pkg/network/types"
+)
+
+// ConnectionAggregator provides functionality for rolling-up datapoints from
+// different connections that refer to the same (client, server) pair
+type ConnectionAggregator struct {
+	data map[ipTuple]portValues
+}
+
+// NewConnectionAggregator returns a new instance of a `ConnectionAggregator`
+func NewConnectionAggregator() *ConnectionAggregator {
+	return &ConnectionAggregator{
+		data: make(map[ipTuple]portValues, 1000),
+	}
+}
+
+type ipTuple struct {
```

Comment:
can't we use struct from `net` package?

Reply:
Are you talking about the `netip.Addr` type? I can only think of downsides in this case, to be honest. I think we're better off using something that matches the IP representation from our `types.ConnectionKey`, since that's what we're interacting with in the code.
Moreover the `netip.Addr` is 50% larger compared to `ipTuple` (24 vs 16 bytes) since it holds a pointer in addition to the128bit field for the IP address